uber/models/attendee.py

```diff
@@ -762,33 +764,53 @@ def merch(self):
         - People with staff badges get a configurable number of staff shirts.
         - Volunteers who meet the requirements get a complementary event shirt
             (NOT a staff shirt).
+
+        If the c.SEPARATE_STAFF_SWAG setting is true, then this excludes staff
+        merch; see the staff_merch property.
         """
         merch = self.donation_swag
 
-        if self.volunteer_event_shirt_eligible:
-            shirt = c.DONATION_TIERS[c.SHIRT_LEVEL]
-            if self.paid_for_a_shirt:
-                shirt = 'a 2nd ' + shirt
-            if not self.volunteer_event_shirt_earned:
-                shirt += (
-                    ' (this volunteer must work at least 6 hours or '
-                    'they will be reported for picking up their shirt)')
+        if self.num_event_shirts_owed == 1 and not self.paid_for_a_shirt:
+            merch.append(c.DONATION_TIERS[c.SHIRT_LEVEL])
+        elif self.num_event_shirts_owed > 1:
+            shirt = 'a 2nd ' + c.DONATION_TIERS[c.SHIRT_LEVEL]
             merch.append(shirt)
 
+        if self.volunteer_event_shirt_eligible \
+                and not self.volunteer_event_shirt_earned:
+            merch[-1] += (
+                ' (this volunteer must work at least 6 hours or '
+                'they will be reported for picking up their shirt)')
+
+        if not c.SEPARATE_STAFF_MERCH:
+            merch.extend(self.staff_merch_items)
+
+        if self.extra_merch:
+            merch.append(self.extra_merch)
+
+        return comma_and(merch)
+
+    @property
+    def staff_merch_items(self):
+        """Used by the merch and staff_merch properties for staff swag items."""
+        merch = []
```

Comment:
This method looks weird as displayed in this diff, since it has code moved here from the ``merch`` property above.  If it's difficult to follow then I recommend opening the file and just reviewing the ``staff_merch_items`` method standalone.